packages/edit-site/src/components/sidebar-navigation-screen-template/index.js

```diff
@@ -98,6 +98,11 @@ export default function SidebarNavigationScreenTemplate() {
 		postId
 	);
 
+	// The absence of a post type in the query params for templates
+	// indicates the user has arrived at the template via the "manage all"
+	// page and the back button should return them to that list page.
+	const backPath =
+		postType !== 'wp_template' ? `/${ postType }/all` : `/${ postType }`;
```

Comment:
Ensure that we go back to the right screen after deleting a custom template or template part:
1. When we're arriving from the site editor sidebar, then back to that. 
2. If we're arriving from the manage all screen, stay on that (`/all`)